hack/lint-go.sh

```diff
@@ -14,5 +14,5 @@ then
     echo "ERROR: golangci-lint could not be found, Go linting aborted. To install it visit https://golangci-lint.run/usage/install/#local-installation"
     exit 1
 else
-    golangci-lint run
+    golangci-lint run --verbose
```

Comment:
Just to check my understanding: Is this change and the change in `.golangci.yml` in fact separate to the main thrust of this PR and actually just a nice general improvement for us?

Not opposed to making things more verbose but just checking this is intentional as I didn't expect it based on the PR description.

Reply:
@mociarain Yes, the changes in `hack/lint-go.sh` and `golangci.yml` are simply general improvements that don't have to be associated with this PR.  Upping the verbosity helped me troubleshoot my install of `golangci-lint` and may help us in future pipeline run troubleshooting.